### PR TITLE
Runtime check

### LINKED ISSUE #107: Return to land after time exceeded
# Return to land after time exceeded

## Problem

Per competition rules, we will be given 30 minutes to complete the mission. Vehicle should return to land if we are approaching the end of our time and have not yet completed all mission objectives.

## Solution

Track time from take-off. Assuming it will take a few minutes to return to home and land, should begin this process before 30 minutes has elapsed.

## Additional Notes

An alternative would be for the safety pilot to take over, but this would incur a manual takeover penalty.

### PR DESCRIPTION
**Pull request by DJDMVP**
changes to flight manager to check time and if it reaches 28 minutes returns the drone to home

### CODE REVIEW

**Comment by fallscameron01** [comment]
Not passing `pylint`. See report below:
```
- hook id: pylint
- exit code: 30

************* Module flight.flight
flight/flight.py:62:0: C0301: Line too long (101/100) (line-too-long)
flight/flight.py:242:0: C0301: Line too long (148/100) (line-too-long)
flight/flight.py:253:0: C0301: Line too long (111/100) (line-too-long)
flight/flight.py:25:4: W0107: Unnecessary pass statement (unnecessary-pass)
flight/flight.py:28:0: R0903: Too few public methods (1/2) (too-few-public-methods)
flight/flight.py:145:8: W0707: Consider explicitly re-raising using 'except Exception as exc' and 'raise DroneNotFoundError() from exc' (raise-missing-from)
flight/flight.py:169:11: W0703: Catching too general exception Exception (broad-except)
flight/flight.py:185:8: W0702: No exception type(s) specified (bare-except)
flight/flight.py:213:4: W0702: No exception type(s) specified (bare-except)
flight/flight.py:240:0: R1711: Useless return at end of function or method (useless-return)
flight/flight.py:7:0: C0411: standard import "from multiprocessing import Queue" should be placed before "from mavsdk import System" (wrong-import-order)
flight/flight.py:14:0: C0411: standard import "import time" should be placed before "from mavsdk import System" (wrong-import-order)
************* Module flight_manager
flight_manager.py:97:8: R1732: Consider using 'with' for resource-allocating operations (consider-using-with)
flight_manager.py:99:34: E1101: Instance of 'BaseManager' has no 'Communication' member (no-member)
flight_manager.py:115:8: W1203: Use lazy % formatting in logging functions (logging-fstring-interpolation)
flight_manager.py:116:8: W1203: Use lazy % formatting in logging functions (logging-fstring-interpolation)
flight_manager.py:13:0: C0411: standard import "import time" should be placed before "from logger import init_logger, worker_configurer" (wrong-import-order)
flight_manager.py:13:0: W0611: Unused import time (unused-import)

-----------------------------------
Your code has been rated at 8.66/10
```

File: flight/flight.py [hunk 1]
```diff
@@ -234,3 +235,27 @@ def flight(
...
+    Returns
+    -------
+    None
+    """
+    # gets the current time and compares it to the time the statemachine was started and returns the difference
+    start = time.time()
+    now = time.time()
+    timespan = now - start
+    while timespan > 1680.0:
+        time.sleep(60)
```
Comment (cjhr95) [inline flight/flight.py]: Please label the significance of these two values with a comment

File: flight/flight.py [hunk 2]
```diff
@@ -234,3 +235,27 @@ def flight(
...
+    Keeps track of run time since this function has been called and if the time is greater than 28 minutes in seconds it calls for the drone to land
+
+    Parameters
+    ----------
+    start: float
+        time in seconds when the drone has started
+
+    Returns
+    -------
+    None
```
Comment (cjhr95) [inline flight/flight.py]: If it only returns None, this section of the docstring isn't required

File: flight_manager.py [hunk 3]
```diff
@@ -122,6 +126,12 @@ def run_threads(self, sim: bool) -> None:
                     logging.error("Flight process terminated, restarting")
                     flight_process = self.init_flight(flight_args)
                     flight_process.start()
+                elif time_process.is_alive() is not True:
```
Comment (cjhr95) [inline flight_manager.py:131]: Could you just do `elif not time_process.is_alive():`

**Review by cjhr95 — CHANGES_REQUESTED** [review]
Some labeling stuff

File: flight/flight.py [hunk 4]
```diff
@@ -238,3 +238,28 @@ def flight(
...
+    ----------
+    start: float
+        time in seconds when the drone has started
+    """
+
+    # find the difference between the current time and the time the state machine started
+    start = time.time()
+    now = time.time()
+    timespan = now - start
+    # 1680 is seconds in seconds and that signals that we are almost out of time
```
Comment (cjhr95) [inline flight/flight.py]: I'm assuming this is supposed to be "1680 is 28 minutes in seconds"

**Review by cjhr95 — CHANGES_REQUESTED** [review]
Fix the comment please :)

**Review by cjhr95 — APPROVED** [review]
Go for it, boss

**Pull request merged**